handbook/engineering/search/goals.md

```diff
@@ -4,48 +4,48 @@
 
 Goals help us achieve our [vision](index.md#vision) for search.
 
-### Scalable
+### Mature query language
 
-- **Scale indexed search to 500k repositories**
-   - **Problem:** We have customers who need Sourcegraph to scale to 500k repositories. Some parts of Sourcegraph don't work well at that scale.
-   - **Outcome:** Sourcegraph can search 500k repositories in less than 300ms. This is on par with [grep.app](https://grep.app).
-   - **Planned work:** [Scaling indexed text search to 500k repositories](#4-scaling-indexed-text-search-to-500k-repositories)
+**Problem:** customers choose other search tools over Sourcegraph because they cannot express the queries they care about with Sourcegraph. TODO customer examples.
```

Comment:
The problem was never that customers chose other search tools. It was more about being able to be a competitive solution for customers using other search solutions, who weren't happy with those solutions either. Either way, I believe we've gotten past this point with the bulk of AND/OR query work. In addition to this prior motivation, we had many requests to provide custom search filters to enable different complex queries, which we previously solved by creating the new filters for each. Adding more filters is not a long term solution, and there needs to be a way to construct these complex queries.

```suggestion
**Problem:** The Sourcegraph query language does not support the growing complexity of searches being run by our users.

Examples:

- I want to only show examples from repositories that have been modified in the last 3 months so I see the most recent solutions.
- I want to find repositories that have not been modified in over a year so I can deprecate them.
- TODO: add more examples.
```

Reply:
Great points, thanks!